From 1bbdc017c8f5aab254fa75b1c792b175211e0693 Mon Sep 17 00:00:00 2001
From: Adam Poniatowski <adam@poniatowski.dev>
Date: Sun, 10 Sep 2023 21:02:51 +0200
Subject: [PATCH] Build error fix

added the missing headers that were causing the build failures
---
 src/lib/game/connectionmanager.h         | 1 +
 src/lib/game/data/units/commandodata.cpp | 3 +++
 src/lib/game/data/units/commandodata.h   | 2 ++
 src/lib/utility/string/utf-8.h           | 1 +
 4 files changed, 7 insertions(+)

diff --git a/src/lib/game/connectionmanager.h b/src/lib/game/connectionmanager.h
index 7f1b6aa6f..9fa490fa6 100644
--- a/src/lib/game/connectionmanager.h
+++ b/src/lib/game/connectionmanager.h
@@ -23,6 +23,7 @@
 #include <memory>
 #include <mutex>
 #include <vector>
+#include <stdexcept>
 
 class cNetwork;
 class cSocket;
diff --git a/src/lib/game/data/units/commandodata.cpp b/src/lib/game/data/units/commandodata.cpp
index 07838496b..211a8ef4e 100644
--- a/src/lib/game/data/units/commandodata.cpp
+++ b/src/lib/game/data/units/commandodata.cpp
@@ -27,6 +27,9 @@
 #include "utility/crc.h"
 #include "utility/position.h"
 
+#include <cstdint>
+
+
 //-----------------------------------------------------------------------------
 /** Checks if the target is on a neighbor field and if it can be stolen or disabled */
 //------------------------------------------------------------------------------
diff --git a/src/lib/game/data/units/commandodata.h b/src/lib/game/data/units/commandodata.h
index 09336093a..8b5f2a7bd 100644
--- a/src/lib/game/data/units/commandodata.h
+++ b/src/lib/game/data/units/commandodata.h
@@ -22,6 +22,8 @@
 
 #include "utility/serialization/nvp.h"
 
+#include <cstdint>
+
 class cMapView;
 class cPosition;
 class cUnit;
diff --git a/src/lib/utility/string/utf-8.h b/src/lib/utility/string/utf-8.h
index d57e778b2..16e1c727f 100644
--- a/src/lib/utility/string/utf-8.h
+++ b/src/lib/utility/string/utf-8.h
@@ -21,6 +21,7 @@
 #define utility_string_utf8H
 
 #include <string>
+#include <cstdint>
 
 namespace utf8
 {
